### ISSUE TITLE
should options be inputs ?

### ISSUE THREAD

**Issue by xquery** [xproc/notes#20]
we have discussed this before, but there are implications in terms of an API version, worth a re-discuss.

**Comment by htInEdin** [comment]
James Fuller writes:

> we have discussed this before, but there are implications in terms of
> an API version, worth a re-discuss.

I think so.  Or rather, I think all options should be (understood as)
set from pipes, so that the dependency/ordering story is uniform.

## ht

```
   Henry S. Thompson, School of Informatics, University of Edinburgh
  10 Crichton Street, Edinburgh EH8 9AB, SCOTLAND -- (44) 131 650-4440
            Fax: (44) 131 650-4587, e-mail: ht@inf.ed.ac.uk
                   URL: http://www.ltg.ed.ac.uk/~ht/
```

 [mail from me _always_ has a .sig like this -- mail without it is forged spam]

**Comment by xquery** [comment]
to simplify,  would the signature of a step impl in another language be:

```
myLangFunc($options as map,Inputs ...)
```

or 

we provide options as an input

```
myLangFunc(Inputs ...)
```

In previous discussions I sided with the former but I am concerned if we can get a stable signature across languages.

**Comment by ndw** [comment]
I'd be reluctant to take on "stable signatures across languages" as a requirement. You wind up in IDL hell with APIs that suck in every language.

**Comment by xquery** [comment]
yes agreed, I am doing some research in how we might use annotations
